packages/rum/src/index.d.ts

```diff
@@ -105,6 +105,7 @@ declare class Span extends BaseSpan {
 }
 
 interface AgentConfigOptions {
+  apiVersion?: number
```

Comment:
Thanks @marksmithson for the PR. Can you restrict this to either it can be number 2 or 3? Right now the agent sends the events to Intake API `v2` by default and only if its set to 3, we enable payload compresion. 

Also i feel by restricting to 2 or 3, We know we can drop the other one when doing major version changes.